array_test.go

```diff
@@ -808,3 +812,37 @@ func TestArrayEvery(t *testing.T) {
 		array.chain.assertFailed(ts)
 	})
 }
+
+func TestArrayFilter(t *testing.T) {
```

Comment:
Please also add tests for two corner cases:

* original array is empty
* original array is non-empty, but Filter() skips all elements and return empty array

Reply:
Also, in each test, please also check that after Filter() call, original array was not modified (has the same value as when initialized).

Reply:
The second one is still unresolved.